matminer/featurizers/base.py

```diff
@@ -393,3 +392,65 @@ def citations(self):
 
     def implementors(self):
         return list(set(sum([f.implementors() for f in self.featurizers], [])))
+
+
+class StackedFeaturizer(BaseFeaturizer):
+    """Use the output of a machine learning model as features
+
+    For regression models, we use the single output class.
+
+    For classification models, we use the probability for the first N-1 classes where N is the
+    number of classes.
+    """
+
+    def __init__(self, featurizer=None, model=None, name=None, class_names=None):
+        """Initialize featurizer
+
+        Args:
+            featurizer (BaseFeaturizer): Featurizer used to generate inputs to the model
+            model (BaseEstimator): Fitted machine learning model to be evaluated
+            name (str): [Optional] name of model, used when creating feature names
+            class_names (str): Required for classification models, used when creating
+                feature names (scikit-learn does not specify the number of classes for
+                a classifier)
+        """
+
+        self.name = name
+        self.class_names = class_names
+        self.featurizer = featurizer
+        self.model = model
+
+    def _is_classifier(self):
+        """Whether the underlying model is a classifier
+
+        Return:
+            (boolean) whether `self.model` is a classifier
+        """
+        return is_classifier(self.model) or hasattr(self.model, 'predict_proba')
+
+    def featurize(self, *x):
+        # Generate the features
+        # TODO: Explore checking whether features have already been computed. Feature for MultiFeaturizer? -lw
+        features = [self.featurizer.featurize(*x)]
+
+        # Run the model
+        if self._is_classifier():
+            output = self.model.predict_proba(features)[0]
+            return output[:-1]
+        else:
+            return self.model.predict(features)
+
+    def feature_labels(self):
+        name = self.name or ''
+        if self._is_classifier():
+            if self.class_names is None:
```

Comment:
should this check go in the __init__() ?

Pro: avoid a situation where you spent time featurizing the data and then get an error only at the end when calling feature_labels(), forcing you to go back to the beginning
Pro: throw the error right away as soon as you know there might be a problem so the user fixes it right away
Con: *Maybe* you want to allow not setting ``class_names`` for a classifier if someone doesn't care about feature_labels? If this is the case, you can still consider throwing a warning that feature_labels won't work if you don't set class_names

Reply:
(the warning would come at initialization)

Reply:
I like the idea of printing a warning for init() that class names are required for `featurize_dataframe` and `feature_labels` to function, but not throwing an error.